server.go

```diff
@@ -936,3 +756,280 @@ func (s *Server) handlerHistory(w http.ResponseWriter, r *http.Request) {
 		s.l.Error("Error rendering template: %v", err)
 	}
 }
+
+func (s *Server) handleJikan(data *dataAddMovie, w http.ResponseWriter, r *http.Request, sourcelink string) ([]string, error) {
+
+	jikanEnabled, err := s.data.GetCfgBool("JikanEnabled", DefaultJikanEnabled)
+	if err != nil {
+		s.l.Error("Error while retriving config value 'JikanEnabled':\n %v", err)
+		s.doError(
+			http.StatusInternalServerError,
+			"something went wrong :C",
+			w, r)
+		return nil, err
+	}
+
+	s.l.Debug("jikanEnabled: %v", jikanEnabled)
+
+	if !jikanEnabled {
+		s.l.Debug("Aborting Jikan autofill since it is not enabled")
+		data.ErrorMessage = append(data.ErrorMessage, "Jikan API usage was not enabled by the site administrator")
+		return nil, fmt.Errorf("Jikan not enabled")
+	} else {
+		// Get Data from MAL (jikan api)
+		rgx := regexp.MustCompile(`[htp]{4}s?:\/\/[^\/]*\/anime\/([0-9]*)`)
```

Comment:
This `MustCompile` should be in the global scope so it is only built at startup and not every time it's run.

Reply:
Will be fixed in the next commit